sklearn/linear_model/coordinate_descent.py

```diff
@@ -585,7 +599,7 @@ class ElasticNet(LinearModel, RegressorMixin):
     def __init__(self, alpha=1.0, l1_ratio=0.5, fit_intercept=True,
                  normalize=False, precompute=False, max_iter=1000,
                  copy_X=True, tol=1e-4, warm_start=False, positive=False,
-                 random_state=None, selection='cyclic'):
+                 random_state=None, selection='cyclic', check_input=True):
```

Comment:
Other scikit-learn models use `check_input` only as per-method parameters rather than global constructor parameter when used in estimators. I think we should stay consistent with that pattern.